File: server/python/send-template-to-audience.py
import os
from pathlib import Path
from dotenv import load_dotenv
from courier import Courier

load_dotenv(Path(__file__).parent.parent / ".env")

api_key = os.getenv("COURIER_API_KEY")
audience_id = os.getenv("COURIER_SEND_TEMPLATE_TO_AUDIENCE_AUDIENCE_ID")
template_id = os.getenv("COURIER_SEND_TEMPLATE_TO_USER_ID_TEMPLATE_ID")

client = Courier(api_key=api_key)

response = client.send.message(
    message={
        "to": {
            "audience_id": audience_id
        },
        "template": template_id,
    },
)

print(response)

File: server/node/send-template-to-audience.js
import Courier from '@trycourier/courier';
import dotenv from 'dotenv';

dotenv.config({ path: new URL('../.env', import.meta.url) });

const apiKey = process.env.COURIER_API_KEY;
const audienceId = process.env.COURIER_SEND_TEMPLATE_TO_AUDIENCE_AUDIENCE_ID;
const templateId = process.env.COURIER_SEND_TEMPLATE_TO_LIST_TEMPLATE_ID;

const client = new Courier({ apiKey });

const response = await client.send.message({
  message: {
    to: {
      audience_id: audienceId
    },
    template: templateId
  }
});

console.log(response);

